### PR TITLE
add regex validation function

### PR DESCRIPTION
**Pull request by WhileLoop**
add a regex validation helper function and a regex example

### CODE REVIEW

**Comment by WhileLoop** [comment]
Is it possible to apply functions to anything other than response body?

**Comment by benhowes** [comment]
@WhileLoop Thanks, this is a great addition!

I suspect that you might be able to get away without the group matching aspect of this for the most part? Unless I'm missing a case where you need to be able to match the groups individually?

**Comment by WhileLoop** [comment]
I made the group matching useful by allowing them to be saved and reused.

**Comment by benhowes** [comment]
@WhileLoop Sorry for the delay in getting back to you on this. I've been thinking about it a bit and wondering if a nicer way of doing this could be to use "named groups" in regexes. This could mean that we automatically save the contents of the named groups to the name if provided. Un-named groups could be dropped. The re module provides [this](https://docs.python.org/3.6/library/re.html#re.match.groupdict) which would make it pretty easy to implement too.

What do you think?

File: tavern/testutils/helpers.py [hunk 1]
```diff
@@ -90,17 +90,12 @@ def validate_regex(response, expression):
...
     assert match
 
-    full_match = match.group(0)
-    groups = [group for group in match.groups()]
     return {
-        "regex": Box({
-            'full_match': full_match,
-            'groups': groups
-        })
+        "regex": Box(match.groupdict())
```
Comment (benhowes) [inline tavern/testutils/helpers.py]: I reckon they might be okay without being keyed on the dict, i.e. they could just be top level variables? so you can just return the `Box(match.groupdict())`

**Comment by benhowes** [comment]
I think this is looking good @WhileLoop! I think @michaelboulton should have a look too and possibly a few tests of the functionality can be added.

File: tavern/testutils/helpers.py (hunk 1 above)
Comment (WhileLoop) [reply tavern/testutils/helpers.py]: This follows the pattern of the other helpers. Also returning the matches as top level variables increases chance some other existing variable being overwritten inadvertently.

**Comment by michaelboulton** [comment]
The only (minor) issue with this is that it forces people to use named match groups, but I actually like the named groups

**Pull request merged**